pkg/domain/infra/abi/images.go

```diff
@@ -806,3 +817,32 @@ func localPathFromURI(url *url.URL) (string, error) {
 	}
 	return url.Path, nil
 }
+
+func createSignature(m []byte, dockerReference string, mech signature.SigningMechanism, keyIdentity string) ([]byte, error) {
+	newSig, err := signature.SignDockerManifest(m, dockerReference, mech, keyIdentity)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating new signature")
+	}
+	return newSig, nil
+}
+
+func putSignature(sigStoreDir string, signature []byte, instanceDigest digest.Digest) error {
```

Comment:
(Absolutely non-blocking: It might feel a tiny bit more readable to have the “destination” parameters (directory+digest) together, instead of wedging the “content” `signature` parameter in the middle.)